File: dfs_bfs/boj/p2606.py
# 인접 노드 그래프 DFS
from sys import stdin
from collections import deque

computer = int(stdin.readline())
connect = int(stdin.readline())
graph = [[] for _ in range(computer + 1)]
visited = [0] * (computer + 1)
count = 0

def dfs(com):
    global count, visited 
    visited[com] = 1

    for i in graph[com]:
        if visited[i] == 0:
            dfs(i)
            count +=1

def bfs(com):
    global count, visited
    que = deque([com])
    visited[com] = 1

    while que:
        tmp = que.popleft()

        for i in graph[tmp]:
            if visited[i] == 0:
                que.append(i)
                visited[i] = 1
                count += 1

for i in range(connect):
    a, b = map(int, stdin.readline().split())
    graph[a].append(b)
    graph[b].append(a)

# dfs(1)
bfs(1)
# 1번 통해 감염된 컴퓨터 수니까 1번 제외
print(count)

File: dfs_bfs/boj/p2606.java
package dfs_bfs.boj;
import java.util.*;
/**
 * p2606
 */
public class p2606 {
    static int computer, connect, count = 0;
    static int[][] graph;
    static int[] visited;

    static void dfs(int com) {
        visited[com] = 1;

        for(int i= 0; i < graph[com].length; i++) {
            if(graph[com][i] == 1 && visited[i] == 0) {
                dfs(i);
                count++;
            }
        }
    }

    static void bfs(int com) {
        visited[com] = 1;
        Queue<Integer> queue = new LinkedList<>();
        queue.offer(com);

        while(!queue.isEmpty()) {
            int tmp = queue.poll();

            for (int i = 0; i < graph[tmp].length; i++) {
                if (graph[tmp][i] == 1 && visited[i] == 0) {
                    queue.offer(i);
                    visited[i] = 1;
                    count++;
                }
            }
        }

    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        computer = sc.nextInt();
        connect = sc.nextInt();

        graph = new int[computer+1][computer+1];
        visited = new int[computer+1];

        for(int i = 0; i < connect; i++)  {
            int a = sc.nextInt();
            int b = sc.nextInt();

            graph[a][b] = 1;
            graph[b][a] = 1;
        }

        // dfs(1);
        bfs(1);

        System.out.println(count);
        sc.close();
    }
    
}